internal/envoy/run.go

```diff
@@ -64,8 +64,10 @@ func (r *Runtime) Run(ctx context.Context, args []string) (err error) { //nolint
 	}
 
 	waitCtx, waitCancel := context.WithCancel(ctx)
-	sigCtx, sigCancel := signal.NotifyContext(waitCtx, syscall.SIGINT, syscall.SIGTERM)
 	defer waitCancel()
+
+	sigCtx, sigCancel := signal.NotifyContext(waitCtx, syscall.SIGINT, syscall.SIGTERM)
+	defer sigCancel()
```

Comment:
I thought calling CancelFunc multiple times would result in panics but I confirmed it is not. Thanks

> // After the first call, subsequent calls to a CancelFunc do nothing.

Reply:
thanks for researching this!